client/src/pages/EditProfile.js

```diff
@@ -57,11 +58,40 @@ function EditProfile() {
         }
     }
 
-    const updateUIState = (data) => {
+    const updateUIState = async (data) => {
+        console.log('data pipeline: ', data.pipeline)
         setFirstName(data.firstName)
         setLastName(data.lastName)
         setSchool(data.school)
         setAnonymous(data.anonymous)
+        for (const pipeline of data.pipeline) {
+            const response = await fetch(
+                `${HOST}/api/company/get/${pipeline.company}`,
+                {
+                    method: 'GET',
+                    headers: {
+                        'Content-Type': 'application/json', // Specify the content type as JSON
+                    },
+                }
+            )
+            if (!response.ok) {
+                // Check if the response has JSON content
+                if (
+                    response.headers
+                        .get('content-type')
+                        ?.includes('application/json')
+                ) {
+                    const errorData = await response.json()
+                    throw new Error(`${errorData.error}`)
+                } else {
+                    throw new Error(`HTTP error! Status: ${response.status}`)
+                }
+            } else {
+                const data = await response.json()
+                pipeline.company =
+                    data.name.charAt(0).toUpperCase() + data.name.slice(1)
```

Comment:
I'm guessing this change was due to a backend change I'm unaware of, so if it works as intended then I'd just need a quick summary or something 😭. I just wanted to make sure, if this logic checks out then we're good to go.

Reply:
Yeah

Reply:
Basically we store all the companies as "apple" instead of "Apple" so that's the reasoning, but I feel like we should just make it so it's converted back in the backend and returned as "Apple" @Lewin-B